### PR TITLE
Move allow guests preferences to conversation info

### PR DESCRIPTION
**Pull request by timkrueger**
<table>
  <tr>
    <th></th>
    <th></th>
    <th></th>
  <tr>
</table>

### CODE REVIEW

**Comment by timkrueger** [comment]
@AlvaroBrey Is it possible to set the correct colors in the AlertDialog?

**Comment by AndyScherzinger** [comment]
> Is it possible to set the correct colors in the AlertDialog?

@timkrueger yes it is see 
https://github.com/nextcloud/talk-android/blob/b90894519592e401719729d7a2efa14ec654fdb2/app/src/main/java/com/nextcloud/talk/controllers/ConversationInfoController.kt#L367-L382

you need to use the Material one instead and then call the coloring methods at the right time(s). Important as seen the the code reference, the buttons needs to be colored _after_ `show()` has been called on the dialog.

**Comment by timkrueger** [comment]
> @timkrueger yes it is, see
> 
> https://github.com/nextcloud/talk-android/blob/b90894519592e401719729d7a2efa14ec654fdb2/app/src/main/java/com/nextcloud/talk/controllers/ConversationInfoController.kt#L367-L382
> 
> you need to use the Material one instead and then call the coloring methods at the right time(s). Important as seen the the code reference, the buttons need to be colored _after_ `show()` has been called on the dialog.

Thank you for your advice @AndyScherzinger

File: app/src/main/java/com/nextcloud/talk/conversation/info/GuestAccessHelper.kt [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+                    as SwitchCompat
+                )
+            guestAccessPasswordSwitch.isChecked = passwordResult.passwordSet && setPassword
+
+            if (passwordResult.passwordIsWeak) {
+                val builder = MaterialAlertDialogBuilder(activity)
+                builder.apply {
+                    setTitle(R.string.nc_guest_access_password_weak_alert_title)
+                    setMessage(passwordResult.message)
+                    setPositiveButton("OK") { _, _ -> }
```
Comment (AndyScherzinger) [inline app/src/main/java/com/nextcloud/talk/conversation/info/GuestAccessHelper.kt:242]: needs to use a translatable string instead of hard-coded "OK"

Comment (AndyScherzinger) [reply app/src/main/java/com/nextcloud/talk/conversation/info/GuestAccessHelper.kt:242]: there should be a common_ok or similar string already

File: app/src/main/java/com/nextcloud/talk/repositories/conversations/ConversationsRepository.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Nextcloud Talk application
+ *
+ * @author Your Name
```
Comment (AndyScherzinger) [inline app/src/main/java/com/nextcloud/talk/repositories/conversations/ConversationsRepository.kt:4]: -> Tim Krüger

File: app/src/main/java/com/nextcloud/talk/repositories/conversations/ConversationsRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,113 @@
+/*
+ * Nextcloud Talk application
+ *
+ * @author Your Name
```
Comment (AndyScherzinger) [inline app/src/main/java/com/nextcloud/talk/repositories/conversations/ConversationsRepositoryImpl.kt:4]: -> Tim Krüger

File: app/src/main/res/layout/dialog_password.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+ Nextcloud Talk application
+
+ @author Your Name
```
Comment (AndyScherzinger) [inline app/src/main/res/layout/dialog_password.xml:5]: -> Tim Krüger

File: app/src/main/res/layout/guest_access_settings_item.xml [hunk 5]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+ Nextcloud Talk application
+
+ @author Your Name
```
Comment (AndyScherzinger) [inline app/src/main/res/layout/guest_access_settings_item.xml:4]: -> Tim Krüger

**Review by AndyScherzinger — APPROVED** [review]
Just some minor cosmetics

**Pull request merged**